api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/merge/Dot.kt

```diff
@@ -0,0 +1,126 @@
+package org.jetbrains.kotlinx.dl.api.core.layer.merge
+
+import org.tensorflow.Operand
+import org.tensorflow.op.Ops
+import org.tensorflow.op.Scope
+import org.tensorflow.op.core.*
+import org.tensorflow.op.linalg.MatMul
+import org.tensorflow.op.linalg.Transpose
+import org.tensorflow.op.math.Mul
+import org.tensorflow.op.math.Rsqrt
+import org.tensorflow.op.math.Square
+
+public class Dot(
+    val axis: IntArray,
+    val normalize: Boolean = false,
+    name: String = ""
+) : AbstractMerge("Dot", name) {
+
+    constructor(axis: Int) : this(axis = IntArray(2) { axis })
+
+    private fun l2Normalize(scope: Scope?, x: Operand<Float>, axis: IntArray?): Operand<Float> {
```

Comment:
Add documentation KDocs for this function, move out of class and make public (it could be reused as a building block)

Reply:
`l2normalize` and `batchdot` are present in linalg and backend in respectively in the official Keras.Where should i refactor them out of here? Or should i just refactor them out of this class ad put it in `Dot.kt` only?

Reply:
I agree that there is no best place for them at this moment (and other missed, implemented-in-Keras only parts). Could you please just refactor them out of class and put it in Dot.kt like you suggested